src/Core/src/Core/Extensions/ITextInputExtensions.cs

```diff
@@ -46,9 +46,6 @@ public static bool TextWithinMaxLength(this ITextInput textInput, string? text,
 #if __ANDROID__
 		public static void UpdateText(this ITextInput textInput, Android.Text.TextChangedEventArgs e)
 		{
-			if (e.BeforeCount == 0 && e.AfterCount == 0)
-				return;
```

Comment:
In https://github.com/jonathanpeppers/maui/commit/85b56990c6e2d97a1d96d1bf2decc91e4231f7fb, it doesn't say why this check was added, but it was causing test failures when MaxLength is set to 0.

If you had an `Entry` with text, then set `MaxLength=0` it wasn't updating `Entry.Text` after the platform fired the `TextChanged` event.